### ISSUE TITLE
Columns from views are not filtered out

### ISSUE THREAD

**Issue by Oreilles** [knex/knex-schema-inspector#48]
`tableInfo()` filters the views out:
https://github.com/knex/knex-schema-inspector/blob/01929fb70bc2391309a2e283ba5ed198f1c4769a/lib/dialects/postgres.ts#L124
But `columns()` doesn't, therefore returning columns with tables we can't get info on.

Maybe it would be preferable to add a `is_view` attribute to the `Table` type.

**Comment by rijkvanzanten** [comment]
Should we include views under regular columns, or have a dedicated `.views()` method? 🤔

**Comment by Oreilles** [comment]
> Should we include views under regular columns, or have a dedicated `.views()` method? 🤔

Then you'd have to set a filter for columns in order not to encounter the problem I raised.
It's probably easier to just add a view attribute 🙂

**Comment by rijkvanzanten** [comment]
True, we'd have to make a change to fix this issue regardless, just trying to figure out what the ideal long-term solution is 🙂 

We either treat "views" as regular tables/columns, but with a flag `is_view`, or we treat them as a separate entity and return them under `.views`. 

I guess it's most common to see/interact with them in the same bucket as regular tables, so lets go with that `is_view` flag 👍🏻

**Comment by rijkvanzanten** [comment]
Ref https://github.com/directus/directus/issues/5815

**Comment by rijkvanzanten** [comment]
@Oreilles For the time being, I'll filter out the views, so it matches the tableInfo method 👍🏻  Then we'll revisit and add proper views support both in here and `knex` itself (ref https://github.com/knex/knex/issues/1626)

**Comment by azrikahar** [comment]
> and `knex` itself (ref [knex/knex#1626](https://github.com/knex/knex/issues/1626))

Just as an update: this was recently resolved by https://github.com/knex/knex/pull/4748

**Comment by rijkvanzanten** [comment]
@azrikahar Knex itself implemented support for views, but that doesn't mean the knex-schema-inspector correctly handles views when querying columnInfo 🙂

**Comment by azrikahar** [comment]
Yea I think what I mean is that the Knex half is there, so you guys can look into the knex-schema-inspector half (since it seems like you're waiting for this), which is why I only quoted the last few letters of your sentence involving Knex only 😅 Apologies if that wasn't clear enough (or if I even misunderstand this latest message 😵)
